ckanext/iati_generator/tests/test_action_iati_file_list.py

```diff
@@ -0,0 +1,145 @@
+from types import SimpleNamespace
+import pytest
+from ckan.tests import helpers, factories
+from ckan.plugins import toolkit
+from ckanext.iati_generator.models.enums import IATIFileTypes
+
+
+@pytest.fixture
+def setup_data():
+    """Crea usuarios, org, dataset y resource base — con tokens+headers listos."""
+    obj = SimpleNamespace()
+
+    # Users + tokens
+    obj.sysadmin = factories.SysadminWithToken()
+    obj.sysadmin["headers"] = {"Authorization": obj.sysadmin["token"]}
+    obj.user_admin = factories.UserWithToken()
+    obj.user_admin["headers"] = {"Authorization": obj.user_admin["token"]}
+    obj.user_editor = factories.UserWithToken()
+    obj.user_editor["headers"] = {"Authorization": obj.user_editor["token"]}
+    obj.user_member = factories.UserWithToken()
+    obj.user_member["headers"] = {"Authorization": obj.user_member["token"]}
+
+    # Org con roles
+    obj.org = factories.Organization(
+        users=[
+            {"name": obj.user_admin["name"], "capacity": "admin"},
+            {"name": obj.user_editor["name"], "capacity": "editor"},
+            {"name": obj.user_member["name"], "capacity": "member"},
+        ]
+    )
+
+    # Dataset + resource
+    obj.pkg = factories.Dataset(owner_org=obj.org["id"])
+    obj.res = factories.Resource(
+        package_id=obj.pkg["id"],
+        format="CSV",
+        url_type="upload",
+        url="test.csv",
+        name="test.csv",
+    )
+    return obj
+
+
+@pytest.mark.usefixtures('with_plugins', 'clean_db')
+class TestIatiFileListAction:
+
+    def _create_iati_file(self, user_name, resource_id,
+                          file_type=IATIFileTypes.ORGANIZATION_MAIN_FILE.value,
+                          is_valid=True, namespace="iati-xml"):
+        """Crea un IATIFile usando la acción pública (asegura validaciones)."""
+        context = {"user": user_name}
+        return helpers.call_action(
+            "iati_file_create",
+            context=context,
+            resource_id=resource_id,
+            file_type=file_type,
+            namespace=namespace,
+            is_valid=is_valid,
+        )
+
+    # ---- helpers for calling actions
+    def _call_action_as_user(self, action, user_name, **data):
+        """Call an action with a specific user context."""
+        context = {"user": user_name}
+        return helpers.call_action(action, context=context, **data)
+
+    # ---------- tests ----------
+    def test_list_returns_items_and_fields(self, setup_data):
+        """Test that the action returns IATI files with expected fields."""
+
+        # Crear 1 IATIFile
+        self._create_iati_file(setup_data.sysadmin["name"], setup_data.res["id"])
+
+        # Sysadmin consulta la acción
+        result = self._call_action_as_user("iati_file_list", setup_data.sysadmin["name"])
+
+        assert result["count"] >= 1
```

Comment:
```suggestion
        assert result["count"] == 1
```
Be precise, we clean the DB on each test.
Test for exact numbers